src/bootstrap/setup.rs

```diff
@@ -21,6 +21,81 @@ pub enum Profile {
     Library,
     Tools,
     User,
+    None,
+}
+
+/// The other steps that may be performed by setuo
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub enum Subcommand {
+    /// Installs `src/etc/pre-push.sh` as a Git hook
+    Hook,
+    /// Sets up or displays `src/etc/vscode_settings.json`
+    Vscode,
+    /// Creates a toolchain link for stage1 using `rustup`
+    Link,
+}
+
+impl FromStr for Subcommand {
+    type Err = &'static str;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "hook" => Ok(Self::Hook),
+            "vscode" => Ok(Self::Vscode),
+            "link" => Ok(Self::Link),
+            _ => Err("invalid setup subcommand"),
+        }
+    }
+}
+
+impl Step for Subcommand {
+    type Output = ();
+    const DEFAULT: bool = true;
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.alias("hook").alias("vscode").alias("link")
```

Comment:
This looks pretty surprising to me - why can't each of the variants be their own Step? It seems like we shouldn't need the intermediate step...